src/azure-cli/azure/cli/command_modules/acs/addonconfiguration.py

```diff
@@ -347,6 +365,9 @@ def ensure_container_insights_for_monitoring(
     if not addon.enabled:
         return None
 
+    if (not is_private_cluster or not aad_route) and ampls_resource_id is not None:
+        raise ArgumentUsageError("--ampls_resource_id can only be used with private cluster in MSI mode.")
```

Comment:
Please check all similar mistakes
```suggestion
        raise ArgumentUsageError("--ampls-resource-id can only be used with private cluster in MSI mode.")
```